app/assets/javascripts/poller.js

```diff
@@ -5,15 +5,60 @@
     init: function() {
       this.$poller = $('.js-poller');
       this.$messageForm = $('.js-message-form');
+      this.pollingTimeout = 1000 * 60 * 60; // 1 hour
 
       if (this.$poller.length && this.$messageForm.length) {
         this.setLast(this.timestamp());
+        this.bindEvents();
         this.start();
       }
     },
 
+    getHiddenProp: function() {
+      var prefixes = ['webkit', 'moz', 'ms'];
+
+      if ('hidden' in document) return 'hidden';
+
+      for (var i = 0; i < prefixes.length; i++) {
+          if ((prefixes[i] + 'Hidden') in document) {
+            return prefixes[i] + 'Hidden';
+          }
+      }
+
+      return null;
+    },
+
+    bindEvents: function() {
+      var prop = this.getHiddenProp();
+
+      if (prop) {
+        $(document).on(
+          prop.replace(/[H|h]idden/, '') + 'visibilitychange',
```

Comment:
Not 100% happy with this - looks a bit ugly - from here: https://www.html5rocks.com/en/tutorials/pagevisibility/intro/

Maybe something cleaner?